### ISSUE TITLE
[BUG] Documentation about config_opts accepting an MFA is incorrect?

### ISSUE THREAD

**Issue by patmaddox** [akoutmos/prom_ex#158]
**Describe the bug**

[Documentation for `config_opts`](https://github.com/akoutmos/prom_ex/blame/611912ce63d4122fedfe8b616ed83cb88521f90f/lib/prom_ex/config.ex#L157) says: "This option can either accept an MFA that will return a string of the full path where the YAML configuration file is, or a keyword list with options so that PromEx can generate a config file for you."

Looking at [the code](https://github.com/akoutmos/prom_ex/blame/611912ce63d4122fedfe8b616ed83cb88521f90f/lib/prom_ex/config.ex#L318) though, it seems pretty clear that it only supports a keyword list - not an MFA.

Perhaps I'm overlooking something?

**To Reproduce**

```elixir
config :my_app, MyApp.PromEx,
  grafana_agent: [
    config_opts: {SomeModule, :foo, []}
  ]
```

Result:

```
** (Mix) Could not start application myapp: MyApp.Application.start(:normal, []) returned an error: shutdown: failed to start child: MyApp.PromEx
    ** (EXIT) an exception was raised:
        ** (FunctionClauseError) no function clause matching in Keyword.get/3
            (elixir 1.13.4) lib/keyword.ex:352: Keyword.get({SomeModule, :foo, []}, :scrape_interval, "15s")
```

**Expected behavior**

It starts.

**Environment**

- Elixir version: Elixir 1.13.4
- Erlang/OTP version: Erlang/OTP 24

**Additional context**

fwiw I'm not quite convinced of the usefulness of the MFA option. You end up having to provide a lot of keys that `PromEx.Config.extract_opts_for_config/1` sets defaults for.

I think it's probably more useful to specify a custom template when rendering.

**Comment by akoutmos** [comment]
That is definitely a miss on my part. It's something I wanted to implement but haven't gotten around to it. Looks like I put the cart before the horse on the docs though. Is this something that you would like to tackle? It shouldn't take very long to implement.

**Comment by patmaddox** [comment]
> Is this something that you would like to tackle?

I don't think so. We started off by defining our config, assuming the MFA thing would work. The args to MFA essentially duplicate the normal config, but also require adding all the missing keys that PromEx sets by default.

Instead, we found it quite elegant to provide a different template file, as in #159

**Comment by yordis** [comment]
@patmaddox do you mind either,

A) Fix the documentation
B) Add support for the MFA?

**Comment by patmaddox** [comment]
Do I mind if you / someone does it? Not at all :) @akoutmos makes the decisions around here though.

**Comment by yordis** [comment]
@patmaddox, my apologies if you felt negative emotions from my comment, @akoutmos needs extra hands since he has other priorities and duties, so I am doing my best to support him as much as possible.

Please don't take it personally.
